components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -87,261 +79,182 @@ public Secret addSecret(String secretTypeName, Secret secret) throws SecretManag
 
-        validateSecretTypeCreateRequest(secretType);
-        String secretTypeID = generateUniqueID();
-        secretType.setId(secretTypeID);
-        getSecretDAO().addSecretType(secretType);
-
-        if (log.isDebugEnabled()) {
-            log.debug("Secret type: " + secretType.getName() + " successfully created with the id: "
-                    + secretType.getId());
+        if (StringUtils.isEmpty(secret.getSecretId()) || StringUtils.isEmpty(secret.getSecretValue())) {
+            throw handleClientException(ErrorMessages.ERROR_CODE_SECRET_UPDATE_REQUEST_INVALID, null);
         }
-        return new SecretType(
-                secretType.getName(),
-                secretType.getId(),
-                secretType.getDescription()
-        );
-    }
-
-    @Override
-    public SecretType replaceSecretType(SecretType secretType) throws SecretManagementException {
 
-        validateSecretTypeReplaceRequest(secretType);
-        String secretTypeID;
-        secretTypeID = retrieveOrGenerateSecretTypeId(secretType.getName());
-        secretType.setId(secretTypeID);
-        getSecretDAO().replaceSecretType(secretType);
-        if (log.isDebugEnabled()) {
-            log.debug("Secret type: " + secretType.getName() + " successfully replaced with the id: "
-                    + secretType.getId());
+        if (secret.getSecretValue() != null) {
+            validateSecretValue(secret.getSecretValue());
         }
-        return new SecretType(
-                secretType.getName(),
-                secretType.getId(),
-                secretType.getDescription()
-        );
-    }
-
-    @Override
-    public SecretType getSecretType(String secretTypeName) throws SecretManagementException {
-
-        validateSecretTypeRetrieveRequest(secretTypeName);
-        SecretType secretType = getSecretDAO().getSecretTypeByName(secretTypeName);
-        if (secretType == null || secretType.getId() == null) {
-            if (log.isDebugEnabled()) {
-                log.debug("Secret Type: " + secretTypeName + " does not exist.");
-            }
-            throw handleClientException(ERROR_CODE_SECRET_TYPE_DOES_NOT_EXISTS, secretTypeName);
+        if (secret.getDescription() != null) {
+            validateSecretDescription(secret.getDescription());
         }
+        if (StringUtils.isEmpty(secret.getTenantDomain())) {
+            secret.setTenantDomain(getTenantDomain());
+        }
+        this.getSecretDAO().updateSecret(secret, getTenantId());
         if (log.isDebugEnabled()) {
-            log.debug("Secret type: " + secretType.getName() + " retrieved successfully.");
+            log.debug(secret.getSecretId() + " secret updated successfully.");
         }
-        return secretType;
+        return secret;
     }
 
     @Override
-    public void deleteSecretType(String secretTypeName) throws SecretManagementException {
+    public Secret updateSecretValue(String secretType, String secretId, String value)
```

Comment:
do we need a dedicated method for this? update secret should be able to achieve this

Reply:
This method is needed for internal operations. Internal components should be able to change secret value without retrieving the secret first.
@senthalan WDYT?

Reply:
yes. the internal components which use the secret management(IDP secret) should be able to update the secret value without pulling up the secret object from the database. The key of the secret will be stored at the component level and during the update, those components will only update the value.